src/components/Calculator.js

```diff
@@ -1,74 +1,101 @@
 import '../calculator.css';
+import Big from 'big.js';
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
+import operate from '../logic/operate';
 
-const Calculator = () => (
-  <main className="calculator">
-    <header className="calc-output">0</header>
+const Calculator = () => {
+  const [state, setState] = useState({ total: null, next: null, operation: null });
 
-    <div className="calc-grid">
-      <section className="calc-numbers">
-        <button id="btn-ac" className="btn" type="button">
-          AC
-        </button>
-        <button id="btn-minus-plus" className="btn" type="button">
-          +/-
-        </button>
-        <button id="btn-percentage" className="btn" type="button">
-          %
-        </button>
-        <button id="btn-one" className="btn" type="button">
-          1
-        </button>
-        <button id="btn-two" className="btn" type="button">
-          2
-        </button>
-        <button id="btn-three" className="btn" type="button">
-          3
-        </button>
-        <button id="btn-four" className="btn" type="button">
-          4
-        </button>
-        <button id="btn-five" className="btn" type="button">
-          5
-        </button>
-        <button id="btn-six" className="btn" type="button">
-          6
-        </button>
-        <button id="btn-seven" className="btn" type="button">
-          7
-        </button>
-        <button id="btn-eight" className="btn" type="button">
-          8
-        </button>
-        <button id="btn-nine" className="btn" type="button">
-          9
-        </button>
-        <button id="btn-zero" className="btn" type="button">
-          0
-        </button>
-        <button id="btn-period" className="btn" type="button">
-          .
-        </button>
-      </section>
+  const handleClick = (buttonName) => {
+    const newState = calculate(state, buttonName);
+    setState(newState);
+  };
 
-      <section className="calc-operators">
-        <button id="btn-four" className="btn" type="button">
-          /
-        </button>
-        <button id="btn-four" className="btn" type="button">
-          *
-        </button>
-        <button id="btn-four" className="btn" type="button">
-          -
-        </button>
-        <button id="btn-four" className="btn" type="button">
-          +
-        </button>
-        <button id="btn-four" className="btn" type="button">
-          =
-        </button>
-      </section>
-    </div>
-  </main>
-);
+  const handleEquals = () => {
+    const { total, next, operation } = state;
+    if (total && next && operation) {
+      const result = operate(Big(total), Big(next), operation);
+      setState({ total: result.toString(), next: null, operation: null });
+    }
+  };
```

Comment:
- [ ] I noticed that when I perform a `modulus(%)` operation and press the equals sign, it doesn't give me the remainder as the result. It doesn't give any results also. Remember that the logic in the calculate function is whole since it uses the `operate` function and covers all scenarios including the modulus operation. I would advise you to reconsider using the `operate` function directly in this component. 👍 😎